upgrades/upgradevalidation/validation.go

```diff
@@ -249,6 +249,44 @@ func checkForCharmStoreCharms(_ string, _ StatePool, st State, _ Model) (*Blocke
 	return nil, nil
 }
 
+func checkForCharmsWithNoManifest(_ string, _ StatePool, st State, _ Model) (*Blocker, error) {
+	curls, err := st.AllCharmURLs()
+	if errors.Is(err, errors.NotFound) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+	result := set.NewStrings()
+	for _, curl := range curls {
+		if curl == nil {
+			return nil, errors.New("malformed charm in database with no URL")
```

Comment:
**question:** Why this malformed exception cause an error instead of being ignored + logged (warning or error) ?

What is fundamentally different from having not uploaded/placeholder charms ?

Do we actually want to prevent migration of malformed charm, or is is just a check to avoid nilpointer exception later ?